lib/Sapi.php

```diff
@@ -89,9 +89,28 @@ public static function sendResponse(ResponseInterface $response)
         if (null !== $contentLength) {
             $output = fopen('php://output', 'wb');
             if (is_resource($body) && 'stream' == get_resource_type($body)) {
-                if (PHP_INT_SIZE !== 4) {
+                if (PHP_INT_SIZE > 4) {
                     // use the dedicated function on 64 Bit systems
-                    stream_copy_to_stream($body, $output, (int) $contentLength);
+                    // a workaround to make PHP more possible to use mmap based copy, see https://github.com/sabre-io/http/pull/119
+                    $left = (int) $contentLength;
+                    // copy with 4MiB chunks
+                    $chunk_size = 4 * 1024 * 1024;
+                    stream_set_chunk_size($output, $chunk_size);
+                    // If this is a partial response, flush the beginning bytes until the first position that is a multiple of the page size.
+                    $contentRange = $response->getHeader('Content-Range');
```

Comment:
how common are Range-Requests here? is it sufficient for your reported use case to use this hack in the simpler form your submitted initially?

Reply:
It is not common for downloading a file, but for streaming a video. Regarding to the performance, I found it is also beneficial to downloading a file on HDD: from 70-80 MB/s to 110-120 MB. But I think it might be because of a larger chunk size (4 MiB) is used in this PR, while manually copy implementation in `stream_copy_to_stream` only uses a 8KiB chunk (https://github.com/php/php-src/blob/00d2b7a3d28d29383450fc2607c7d4fb91d57a4a/main/streams/streams.c#L1469 and https://github.com/php/php-src/blob/00d2b7a3d28d29383450fc2607c7d4fb91d57a4a/main/streams/php_streams_int.h#L49).